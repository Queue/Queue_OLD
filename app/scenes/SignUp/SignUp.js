//
// SignUp Scene

import React, { Component } from 'react';
import styles from './styles';
import {View,  Text} from 'react-native';

// Components
import {
  PrimaryButton,
  EmailField,
  PasswordField,
  TextButton
} from '../../components';

// So the keyboard doesnt get in the way
import KeyboardSpacer from 'react-native-keyboard-spacer';

// Data
import Data from '../../lib/data';

// Colors
import Colors from '../../lib/colors';

// Router Actions see ../../lib/navigation
import { Actions } from 'react-native-router-flux'

// Dismiss keyboard
import dismissKeyboard from 'react-native-dismiss-keyboard';

import Common from '../../lib/common';

export default class SignUp extends Component {
  constructor(props) {
    super(props);

    this.state = {
      // Focus bool
      focus: false,

      // Form fields state
      emailText: '',
      passwordText: '',
      unfocus: true,

      // Color
      emailColor: Colors.primaryForeground,
      passwordColor: Colors.primaryForeground
    };

    // Bind all functions
    this.createUser = this.createUser.bind(this);
  }

  createUser() {
    if (Common.validateEmail(this.state.email)) {
      if (this.state.passwordText !== '') {

        // Create a new user
        Data.firebase.auth().createUserWithEmailAndPassword(this.state.emailText, this.state.passwordText).catch(
          (error) => {
            if (error) {
              Common.error(error.code, error.message);
            } else {
              let user = Data.firebase.auth().currentUser;
              if (user) {
                user.sendEmailVerification().then(() => {
                  Common.log('emailsent', 'success');
                }, (error) => {
                  Common.log(error.code, error.message);
                });
              }
            }
          }
        );

        dismissKeyboard();

        // Reset fields state
        this.setState({
          emailText: '',
          passwordText: ''
        });

        Actions.DashboardRoute();

      } else {
        console.warn('Enter a password!');
      }
    }
  }

  checkPassLength(text) {
    if (text.length >= 6) {
      this.setState({passwordColor: Colors.primaryForeground});
    } else {
      this.setState({passwordColor: Colors.error});
    }

    this.setState({passwordText: text});
  }

  render() {
    return (
      <View style = {styles.container}>
        <View style = {styles.wrapper}>
          <Text
            style = {styles.brand}>
            Queue
          </Text>
          <EmailField
            placeholder = {'Email'}
            focus={this.state.focus}
            change = {(text) => this.setState({emailText: text})}
            val = {this.state.emailText}
          />
          <PasswordField
            textColor = {this.state.passwordColor}
            placeholder = {'Password'}
            change = {this.checkPassLength.bind(this)}
            val = {this.state.passwordText}
          />
          <PrimaryButton
            name = {'Sign Up'}
            press = {this.createUser}
          />
          <TextButton
            text={'Already have an account?'}
            size = {16}
            press={Actions.SignInRoute}
          />
          <KeyboardSpacer />
        </View>
      </View>
    );
  }

}

// end
//
