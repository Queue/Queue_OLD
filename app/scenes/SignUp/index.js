//
// SignUp index.js

import SignUp from './SignUp';
export default SignUp;
