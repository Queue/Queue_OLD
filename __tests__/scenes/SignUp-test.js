//
// SignUp scene test

import React from 'react';
import SignUp from '../../app/scenes/SignUp';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <SignUp />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
