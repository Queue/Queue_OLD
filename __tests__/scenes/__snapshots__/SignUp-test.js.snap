exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eaeaea",
      "flex": 1,
      "height": 1334,
      "justifyContent": "center",
      "width": 750,
    }
  }>
  <View
    style={
      Object {
        "width": 300,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#48C948",
          "fontFamily": "airstream",
          "fontSize": 142,
          "marginBottom": -40,
          "textAlign": "center",
        }
      }>
      Queue
    </Text>
    <TextInput
      autoCapitalize="none"
      autoFocus={false}
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 2,
            "fontFamily": "Helvetica Neue",
            "height": 40,
            "justifyContent": "center",
            "margin": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          Object {
            "color": "#48C948",
          },
        ]
      }
      value="" />
    <View>
      <TextInput
        autoCapitalize="none"
        editable={undefined}
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 2,
              "color": "#48C948",
              "fontFamily": "Helvetica Neue",
              "height": 40,
              "justifyContent": "center",
              "margin": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "zIndex": 1,
            },
            Object {
              "color": "#48C948",
            },
          ]
        }
        value="" />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "backgroundColor": "transparent",
              "position": "absolute",
              "right": 20,
              "top": 20,
              "zIndex": 9999,
            },
          ]
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#48C948",
            }
          }>
          👁
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits="button"
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#48C948",
          "borderRadius": 4,
          "borderWidth": 1,
          "margin": 15,
          "marginTop": 20,
          "opacity": 1,
          "overflow": "hidden",
          "padding": 8,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007aff",
              "fontFamily": "HelveticaNeue-Medium",
              "fontSize": 17,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            null,
            Array [
              Object {
                "fontSize": 20,
                "fontWeight": "500",
              },
              Object {
                "color": "#48C948",
              },
            ],
            null,
          ]
        }>
        Sign Up
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#48C948",
              "fontSize": 14,
              "letterSpacing": 1,
              "padding": 5,
              "paddingBottom": 15,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Helvetica Neue",
              "fontSize": 16,
            },
          ]
        }>
        Already have an account?
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "right": 0,
          },
          Object {
            "height": 0,
          },
          undefined,
        ]
      } />
  </View>
</View>
`;
